web/src/engine/websites/CoolMic.ts

```diff
@@ -0,0 +1,81 @@
+import { Tags } from '../Tags';
+import icon from './CoolMic.webp';
+import { Chapter, DecoratableMangaScraper, Page, type Manga } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { Fetch, FetchCSS, FetchJSON } from '../platform/FetchProvider';
+import type { Priority } from '../taskpool/DeferredTask';
+import { Exception } from '../Error';
+import { WebsiteResourceKey as R } from '../../i18n/ILocale';
+
+type JsonChapters = {
+    episodes: {
+        id: number,
+        number : string
+    }[]
+}
+
+type APIPages = {
+    image_data?: {
+        path: string
+    }[],
+    signed_cookie?: CookieSigner
+}
+
+type CookieSigner = {
+    'CloudFront-Policy': string,
+    'CloudFront-Signature': string,
+    'CloudFront-Key-Pair-Id': string
+}
+
+@Common.MangaCSS(/^{origin}\/titles\/\d+$/, 'meta[property="og:title"]')
+@Common.MangasNotSupported()
```

Comment:
Consider providing a distinct manga list by utilizing the search (assuming each title has at least one latin letter):
```javascript
// TODO: Provide more clear URL search parameters with better readability
const promises = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('').map(character => fetch(`https://en-search.coolmic.me/search?q=(${character}%7C${character}*)&size=10&start=0&q.options=%7B%22fields%22:[%22title_name%22],%22operators%22:[%22and%22,%22not%22,%22or%22,%22prefix%22,%22precedence%22,%22whitespace%22],%22defaultOperator%22:%22or%22%7D&q.parser=simple&return=_all_fields&sort=_score+desc,+like_vote_count+desc&fq=`, {
  "headers": {
// TODO: Get a valid token
    "x-csrf-token": "jp9EpimI3qPC2a8Yd0RShgyRLaIj3sWQpw5EnFss0HF4c9lUPxbFNWnp7qakTIbzlO_4QUVFOd8DPgs9rR45_A",
    "x-requested-with": "XMLHttpRequest"
  },
}));
// TODO: Process promises and remove duplicates
```